.github/workflows/behaviour.yml

```diff
@@ -1,5 +1,15 @@
 name: Behaviour tests
-on: [ push, pull_request ]
+on:
+  push:
+    # Do not run after merging a pull request
+    branches-ignore:
+      - 'develop'
+      - '**.x'
+  pull_request:
+    # Do not run when only tests are updated
+    paths-ignore:
+      - 'tests/UI/**'
```

Comment:
Ok for `Do not run when only tests are updated`

But I disagree `Do not run after merging a pull request`

Because a PR can break the CI after a merge (for example, if the PR is old and has no conflicting files).